src/main/java/uk/ac/cam/cl/dtg/segue/auth/SegueLocalAuthenticator.java

```diff
@@ -96,22 +96,24 @@ public void setOrChangeUsersPassword(final RegisteredUser userToSetPasswordFor,
 
     @Override
     public RegisteredUser authenticate(final String usersEmailAddress, final String plainTextPassword)
-            throws IncorrectCredentialsProvidedException, NoUserException, NoCredentialsAvailableException,
+            throws IncorrectCredentialsProvidedException, NoCredentialsAvailableException,
             SegueDatabaseException, InvalidKeySpecException, NoSuchAlgorithmException {
 
         if (null == usersEmailAddress || null == plainTextPassword) {
             throw new IncorrectCredentialsProvidedException("Incorrect credentials provided.");
         }
 
         RegisteredUser localUserAccount = userDataManager.getByEmail(usersEmailAddress);
+        LocalUserCredential luc;
         if (null == localUserAccount) {
-            throw new NoUserException("No user found with this email!");
-        }
-
-        LocalUserCredential luc = passwordDataManager.getLocalUserCredential(localUserAccount.getId());
-        if (null == luc || null == luc.getPassword() || null == luc.getSecureSalt()) {
-            log.debug(String.format("No credentials available for this account id (%s)", localUserAccount.getId()));
-            throw new NoCredentialsAvailableException("This user does not have any local credentials setup.");
+            // Default object so hashing & comparison is still called on non-existent accounts for more consistent return time
+            luc = new LocalUserCredential(-1L, "password", "salt", "SegueSCryptv1");
+        } else {
```

Comment:
Question: why is this an if/else? Shouldn't this be just an if? So that both existing and non-existing users follow the same path and it takes the same time for both.